### PR TITLE
add action to create PR-specific apps

### PR DESCRIPTION
**Pull request by mariogiampieri**
## Description

- Testing github actions to create pr-specific apps for testing before merging into main. Action steps include:
    - [x] forking current database and creating a new instance
    - [x] deploying a the `api` app
        - [x] passing the created database keys to api app
        - [x] being able to access db data via api (previously dealing with incorrect `db_url`)
    - [x] deploying the `app` app
        - [x] ensure that app has permissions under cors policy to access tiles / other bucket assets
        - [x] pass frontend app secrets 
    - [x] running any migrations       
    - [x] deleting these resources once the PR is merged in
    - [x] account for updates to pr re: redeploy

## Reviewers

- Primary:
- Secondary:

## Checklist

- [ ] Added/Updated related documentation (if applicable).
- [ ] Added/Updated related unit tests (if applicable).

### CODE REVIEW

**Review by raphaellaude — CHANGES_REQUESTED** [review]
Looks super good. I like how the action skips rebuilding the db + apps if they exist 🤗 . Just a few small comments

File: app/fly.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ primary_region = 'ewr'
 [http_service]
   internal_port = 3000
   force_https = true
-  auto_stop_machines = 'stop'
+  auto_stop_machines = true
   auto_start_machines = true
-  min_machines_running = 1
+  min_machines_running = 2
```
[the remark this answers is not in the document]
Comment (mariogiampieri) [reply app/fly.toml:16]: yup yup, was just testing but agreed it's overkill for a pr branch

File: app/Dockerfile [hunk 2]
```diff
@@ -26,6 +26,17 @@ RUN npm ci --include=dev
...
+# add args for build-time variables from github actions secrets
+ARG NEXT_PUBLIC_API_URL=${NEXT_PUBLIC_API_URL}
+ARG NEXT_PUBLIC_S3_BUCKET_URL=${NEXT_PUBLIC_S3_BUCKET_URL}
+
+ENV NEXT_PUBLIC_API_URL=${NEXT_PUBLIC_API_URL}
+ENV NEXT_PUBLIC_S3_BUCKET_URL=${NEXT_PUBLIC_S3_BUCKET_URL}
+
+# echo env vars to check if they are set
+RUN echo $NEXT_PUBLIC_API_URL
+RUN echo $NEXT_PUBLIC_S3_BUCKET_URL
```
[the remark this answers is not in the document]
Comment (mariogiampieri) [reply app/Dockerfile:38]: good call!

File: app/fly.toml [hunk 3]
```diff
@@ -11,9 +11,9 @@ primary_region = 'ewr'
 [http_service]
   internal_port = 3000
   force_https = true
-  auto_stop_machines = 'stop'
+  auto_stop_machines = true
```
[the remark this answers is not in the document]
Comment (mariogiampieri) [reply app/fly.toml:14]: noted, and fine with having a different approach for prod vs testing later

File: .github/workflows/fly-deploy-pr.yml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                NEXT_PUBLIC_API_URL="https://pr-${{ github.event.number }}-${{ github.repository_owner }}-${{ github.event.repository.name }}-api.fly.dev" \
+                NEXT_PUBLIC_S3_BUCKET_URL=${{secrets.NEXT_PUBLIC_S3_BUCKET_URL}} 
+
+
+            - name: run database migrations
+              run: |
+                flyctl ssh console \
+                --app "pr-${{ github.event.number }}-${{ github.repository_owner }}-${{ github.event.repository.name }}-api" \
+                --command "alembic upgrade head"
+              working-directory: backend
```
[the remark this answers is not in the document]
Comment (mariogiampieri) [reply .github/workflows/fly-deploy-pr.yml:170]: i think that's right- you all did a great job with the backend documentation and people may end up having to futz with specific migrations if they aren't careful per your point. happy to leave for now, maybe worth changing the boilerplate PR text to state clearly what will happen on each PR?

**Review by raphaellaude — APPROVED** [review]
lgtm

**Comment by mariogiampieri** [comment]
Rebased, added some documentation, and successfully tested shattering on the pr app. I manually removed the app so as to not waste resources but we should be good to go now.

**Review by raphaellaude — APPROVED** [review]
lgtm

**Comment by raphaellaude** [comment]
@mariogiampieri I need this to test a feature real quick so I'm going to merge. Ty!

**Pull request merged**